src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -33,6 +34,9 @@ public async Task Invoke(MessageContext messageContext)
                 try
                 {
                     await receivePipeline.Invoke(transportReceiveContext).ConfigureAwait(false);
+
+                    // mark success
+                    consecutiveFailuresCircuitBreaker.Success();
```

Comment:
I would be great if this could be made entirely NoOp depending on the configuration. So if rate limiting is not configured, we are also not calling success and failure on the hot path.